rust/cloud-storage/properties_service/src/api/properties/options/create.rs

```diff
@@ -75,6 +79,27 @@ pub async fn add_property_option(
 ) -> Result<(StatusCode, Json<PropertyOption>), AddPropertyOptionErr> {
     tracing::info!("adding property option");
 
+    // First check if property exists and if it's a system property
+    let property = property_definitions_get::get_property_definition(&context.db, property_uuid)
+        .await
+        .inspect_err(|e| {
+            tracing::error!(
+                property_id = %property_uuid,
+                error = ?e,
+                "failed to fetch property definition"
+            );
+        })?
+        .ok_or(AddPropertyOptionErr::PropertyNotFound)?;
+
+    if property.is_system {
+        tracing::warn!(
+            property_id = %property_uuid,
+            "attempted to add option to system property"
+        );
```

Comment:
if you instrument the handler with `err` then this would automatically log when you return Err(AddPropertyOptionErr::SystemPropertyNotModifiable)